### PR TITLE
DALI Library modularization

### PR DESCRIPTION
**Pull request by jantonguirao**
Signed-off-by: Joaquin Anton <janton@nvidia.com>

#### Why we need this PR?
Need to separate libdali.so to independent modules (core, kernel, pipeline, operators,...)

#### What happened in this PR?
Split libdali.so into separate shared object libraries (libdali_core, libdali_kernels, libdali, libdali_operators)

**JIRA TASK**: [DALI-1086, DALI-1087, DALI-1088, DALI-1089]

### CODE REVIEW

**Comment by jantonguirao** [comment]
!build

**Comment by jantonguirao** [comment]
!build

File: dali/kernels/CMakeLists.txt [hunk 1]
```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-project(kernels)
+project(${dali_kernel_lib})
```
Comment (mzient) [inline dali/kernels/CMakeLists.txt]: These names are intended as target names - please keep the project name literal, so those who maintain this code don't have to chase definitions.

**Comment by jantonguirao** [comment]
Closing in favor of #1384

**Pull request closed without merging**